src/coreclr/jit/optcse.cpp

```diff
@@ -4130,43 +4186,44 @@ void CSE_Heuristic::Initialize()
             continue;
         }
 
-        // enregCount only tracks the uses of integer registers.
-        //
-        // We could track floating point register usage separately
-        // but it isn't worth the additional complexity as floating point CSEs
-        // are rare and we typically have plenty of floating point register available.
-        //
-        if (!varTypeIsFloating(varTyp))
+        unsigned enregCount;
+        unsigned cntAggressiveEnreg;
+        unsigned cntModerateEnreg;
+
+        if (varTypeUsesIntReg(varTyp))
         {
-            enregCount++; // The primitive types, including TYP_SIMD types use one register
```

Comment:
This is the other place where we said we were tracking integer registers only, but then actually were tracking SIMD and Mask values, which pessimizes things for no good reason.